### ISSUE TITLE
Exchange authentification since October 2022

### ISSUE THREAD

**Issue by jyjourdain** [lefcha/imapfilter#258]
Hi,

I'm not able anymore to connect to Exchange…

I have that error in debug mode :
`1011 NO LOGIN failed.`

Is there a link with the fact that Microsoft removed basic authentification ? https://learn.microsoft.com/fr-fr/lifecycle/announcements/basic-auth-deprecation-exchange-online

Thanks!

**Comment by douglas-larocca** [comment]
Yes, you can re-enable temporarily following the steps here: https://techcommunity.microsoft.com/t5/exchange-team-blog/basic-authentication-and-exchange-online-june-2021-update/bc-p/2599824#M31057

**Comment by jyjourdain** [comment]
Thanks @douglas-larocca !

Do you or anybody else know if imapfilter will support the new Microsoft login protocol?

Thanks in advance!

**Comment by lefcha** [comment]
> Do you or anybody else know if imapfilter will support the new Microsoft login protocol?

Doesn't Microsoft Exchange still support XOAUTH2? Ref: https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth

Imapfilter supports XOAUTH2, you just have to generate the string to supply to the IMAP server using external tools. I think this has been discussed in another Issue: https://github.com/lefcha/imapfilter/issues/252

**Comment by dngray** [comment]
You probably want something like [pizauth](https://github.com/ltratt/pizauth?tab=readme-ov-file#microsoft--exchange) or [oama](https://github.com/pdobsan/oama?tab=readme-ov-file#microsoft-organizational-account) to manage this for you.
